### PR TITLE
[Offload] Switch to torch offloader from accelerate

### PR DESCRIPTION
**Pull request by kylesayrs**
## Co-requisites ##
* Compressed Tensors: [[TorchOffloader] Remove Accelerate](https://github.com/vllm-project/compressed-tensors/pull/530)

## Changes ##
* Perform tracing in the `disable_onloading` context (any tensors that are mentioned will be referenced as meta tensors to avoid excess onloading)
* Dispatch model after tracing (not really necessary)
* `dispatch_for_sequential` is now an alias for `offload_model`
* `dispatch_for_generation` is now an alias for `dispatch_model`
* Add a `get_main_device` utility to help with centralized device getting
* `untie_word_embeddings` is now simpler
* Fix fusing test by initializing tensors without gradients

## Testing ##
* https://github.com/neuralmagic/llm-compressor-testing/actions/runs/21348973738
* Optional TODO: add more tracing tests

### CODE REVIEW

File: src/llmcompressor/utils/dev.py [hunk 1]
```diff
@@ -113,28 +115,30 @@ def patch_transformers_logger_level(level: int = logging.ERROR):
...
-    )
-    device_map = infer_auto_device_map(
-        model,
-        dtype=model.dtype,
-        max_memory=max_memory,
-        no_split_module_classes=no_split_module_classes,
-    )
-
-    return dispatch_model(model, device_map=device_map)
+    return dispatch_model(*args, **kwargs)
```
Comment (HDCharles) [inline src/llmcompressor/utils/dev.py:147]: is this just for BC? unclear why do use dispatch for generation over dispatch model

File: src/llmcompressor/pipelines/sequential/helpers.py [hunk 2]
```diff
@@ -535,16 +539,9 @@ def dispatch_for_sequential(model: PreTrainedModel) -> PreTrainedModel:
...
-        offloaded_dispatch(model, execution_device=torch.device("cuda:0"))
-    elif hasattr(torch, "xpu") and torch.xpu.is_available():
-        offloaded_dispatch(model, execution_device=torch.device("xpu:0"))
-    else:
-        logger.warning("CUDA/XPU is not available! Compressing model on CPU instead")
-
-    return model
+    if onload_device is None:
+        onload_device = get_main_device()
+    return offload_model(model, onload_device, offload_device)
```
Comment (HDCharles) [inline src/llmcompressor/pipelines/sequential/helpers.py:547]: Image

File: src/llmcompressor/utils/transformers.py [hunk 3]
```diff
@@ -28,14 +27,9 @@ def untie_word_embeddings(model: PreTrainedModel):
...
-            data = module.weight.data
-        else:
-            data = module._hf_hook.weights_map["weight"]
-
-        requires_grad = module.weight.requires_grad
-        untied_param = Parameter(data.clone(), requires_grad=requires_grad)
-        register_offload_parameter(module, "weight", untied_param)
+        weight = module.weight
+        param = Parameter(weight.data.clone(), requires_grad=weight.requires_grad)
+        module.register_parameter("weight", param)
```
Comment (HDCharles) [inline src/llmcompressor/utils/transformers.py:32]: Image

File: src/llmcompressor/utils/dev.py (hunk 1 above)
Comment (brian-dellabetta) [reply src/llmcompressor/utils/dev.py:147]: what is BC here?

Comment (kylesayrs) [reply src/llmcompressor/utils/dev.py:147]: @brian-dellabetta BC = backwards compatibility

@HDCharles Yes, ish. The purpose of this PR is to support the new offloader implementation with as few changes as possible. All the code cleanup/ deprecation/ removal happens here: #2147

File: src/llmcompressor/utils/transformers.py (hunk 3 above)
Comment (kylesayrs) [reply src/llmcompressor/utils/transformers.py:32]: I know. I'm excited too

**Review by dsikka — APPROVED** [review]
